### PR TITLE
Set up environment once, run tests one at a time

### PR DESCRIPTION
**Pull request by scbedd**
This should fix any CI issues introduced by adjusted test environments.

Test results are now outputting under a directory labelled with the package they ran for. PublishTestResults should be collecting from all these directories combined now.

Also running `black` formatting on the devops_scripts folder.

`logging` module does a lot to make the output readable. 

Currently trying to understand why .coverage isn't actually uploading any results in some situations. We're definitely properly running `--cov` vs `--cov-append` in the correct locations.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: scripts/devops_tasks/setup_execute_tests.py [hunk 1]
```diff
@@ -12,106 +12,176 @@
...
+            "-p",
+            ",".join([os.path.basename(p) for p in targeted_packages]),
+        ],
+        root_dir,
+    )
+
+
+def run_tests(targeted_packages, python_version, test_output_location, test_res):
+    err_results = []
+    # if we are targeting only packages that are management plane, it is a possibility
```
Comment (danieljurek) [inline scripts/devops_tasks/setup_execute_tests.py]: This comment or one like it might belong closer to line 62

Comment (scbedd) [reply scripts/devops_tasks/setup_execute_tests.py]: Good call. Cleaned.

**Comment by scbedd** [comment]
> LGTM.
> 
> The changes to `testResultsFiles` looks like it picks up an extra 5 tests for each of the 8 matrix entries.

Thanks for looking at this! I will just leave the 5 additional tests alone. I will note that this PR _broke_ coverage because of the fact that for _some reason_ `pytest-cov` will only properly _append_ coverage results if there is an already existing `.coverage` file! I have pushed an update that addresses this.

**Pull request merged**